### DISCUSSION TITLE
Support for PRQL?

### DISCUSSION THREAD

**Discussion by richb-hanover** [pawelsalawa/sqlitestudio#4892] [Ideas]
I've been following the [PRQL project ("pronounced prequel")](https://prql-lang.org/) recently. From their website: PRQL is a modern language for transforming data — a simple, powerful, pipelined SQL replacement. The interesting part about PRQL is that it compiles to SQL statements, so it works with any SQL database manager. (_Check it out - it's quite compelling..._)

I am currently using PRQL to write queries, and let the [PRQL Playground](https://prql-lang.org/playground/) or VScode extension compile them to SQL. I then copy/paste the SQL into my SQLite database to view the results. This works, but it is boring - too many steps.

My question to the group: How might I tie these two projects together? What integration points could I investigate? I'd like to type in the PRQL queries and when I'm happy (and it's syntactically correct), press a button to send the resulting SQL query to SQLiteStudio to display the results.

Many thanks.

**Comment by RandomFractals** [comment]
That might be tricky. It's easy to add extension handler to VS Code extension with some params like SQL query to launch VS Code, and trigger an action in specific VS Code extension on activation. 

We also like SQLite Studio @pawelsalawa created. However, what you are asking might be beyond typical desktop apps integration, unless Pawel has some ideas we can explore for this feature request.

You should be able to just copy the generated SQL from PRQL editor in VS Code and use it in SQLite Studio. We believe those features were added to [PRQL VS Code](https://marketplace.visualstudio.com/items?itemName=PRQL-lang.prql-vscode) ext. earlier this year 🙂

**Comment by richb-hanover** [comment]
> You should be able to just copy the generated SQL from PRQL editor in VS Code and use it in SQLite Studio. We believe those features were added to [PRQL VS Code](https://marketplace.visualstudio.com/items?itemName=PRQL-lang.prql-vscode) ext. earlier this year 🙂

Yes, I copy/paste the PRQL-generated SQL code now. This works fine. But it's too many clicks. (I'm lazy)

I just figured that I would give @pawelsalawa a chance to enhance SQLiteStudio with a unique and powerful tool.

PS I had some questions about the PRQL SQL Pro Tools that you also mention. I'll write to you over there. Thanks.

**Reply by RandomFractals** [reply]
yeah, might be easier to just add PRQL support to SQLite Studio. Also interested in @pawelsalawa take on it.

You should try [PRQL Pro Tools](https://github.com/RandomFractals/pro-data-tools/blob/main/prql-tools.md) if you are a heavy VS Code and PRQL user already to save you all those clicks, context switching to another app for DB views, queries run, etc.

**Reply by pawelsalawa** [reply]
Frankly speaking, I barely find any time to work on SQLiteStudio these days and there are many other high-priority items to work on, so I would not count on having this implemented any time soon.

**Comment by richb-hanover** [comment]
Thank you for the clear guidance. Good luck on all your other projects!
